spack/package.py

```diff
@@ -31,6 +33,7 @@ class GribUtil(CMakePackage):
     depends_on("w3emc precision=4,d", when="^w3emc@2.10:")
     depends_on("w3nco", when="@:1.2.3")
     depends_on("g2")
+    depends_on("g2@3.4.9:", when="@1.4:")
```

Comment:
Not sure how much it matters, but 1.3 depended on g2-3.4.8.

So I think you want to add:
 depends_on("g2@3.4.8:", when="@1.3:")